decorator.rb

```diff
@@ -0,0 +1,12 @@
+require './nameable'
+# A Decorator class
+class Decorator < Nameable
+  def initialize(nameable)
+    super(nameable)
+    @nameable = nameable
+  end
```

Comment:
- [ ] The nameable is expected to have a setter and getter as follows: `attr_accessor :nameable`. Kindly add that before your initialize method. 👍🏽